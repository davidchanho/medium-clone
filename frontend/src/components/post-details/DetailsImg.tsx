import React, { useState } from "react";
import { Modal } from "react-bootstrap";
import { IPost } from "../../types";
import styles from "./DetailsImg.module.scss";

function DetailsImg({ image }: IPost) {
  const [isOpen, setIsOpen] = useState(false);

  const toggleOpen = () => {
    setIsOpen(!isOpen);
  };

  return (
    <>
      <img src={image} className={styles.detailsImg} onClick={toggleOpen} />

      <Modal show={isOpen} className={styles.detailsModal}>
        <Modal.Body>
          <img src={image} className={styles.detailsImg} onClick={toggleOpen} />
        </Modal.Body>
      </Modal>
    </>
  );
}

export default DetailsImg;
