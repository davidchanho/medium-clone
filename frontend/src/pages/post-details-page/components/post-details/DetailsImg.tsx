import React, { useState } from "react";
import { IPost } from "../../../../types";

function DetailsImg({ title, image }: IPost) {
  const [isOpen, setIsOpen] = useState(false);

  const toggleOpen = () => {
    setIsOpen(!isOpen);
  };

  return (
    <>
      <img src={image} alt={title} onClick={toggleOpen} />

      <div>
        <div>
          <img src={image} alt={title} onClick={toggleOpen} />
        </div>
      </div>
    </>
  );
}

export default DetailsImg;
