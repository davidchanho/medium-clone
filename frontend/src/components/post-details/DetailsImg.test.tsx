import React from "react";
import renderer from "react-test-renderer";
import { testPost } from "../../tests";
import DetailsImg from "./DetailsImg";

describe("DetailsImg", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<DetailsImg post={testPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
