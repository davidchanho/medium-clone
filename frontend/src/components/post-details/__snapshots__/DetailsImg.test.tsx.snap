// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsImg renders correctly 1`] = `
Array [
  <img
    alt="test"
    className="w-100"
    onClick={[Function]}
    src="test"
  />,
  <div>
    <div>
      <img
        alt="test"
        onClick={[Function]}
        src="test"
      />
    </div>
  </div>,
]
`;
